src/AlarmClock.h

```diff
@@ -56,87 +67,98 @@ template<typename Duration> class AlarmClock {
 
 protected:
 
-   void AlarmClockThread() {
-      SleepTimeIsBelow500ms() ? AlarmClock::SleepForFullAmount() : AlarmClock::SleepInIntervals();
-   }
-  
-   void StopBackgroundThread() {
-      mExpired.store(true);
-      mExited.wait();
-   }
+   void AlarmClockInterruptableThread() {
+      do {
+         // Call the sleep function
+         unsigned int retVal = mSleepFunction(kSleepTimeUsCount);
 
-   bool SleepTimeIsBelow500ms() {
-      return Duration(kSleepTime) <= microseconds(kSmallestIntervalInUS);
-   }
+         if (retVal == 0) {
+            // Expired normally, should increment mExpired
+            mExpired++;
```

Comment:
mExpired is an atomic, right? Hen it's better to do pre increment since that is one atomic operation

Post increment is multiple operations